### DISCUSSION TITLE
mem.Dynamic_Pool feels misnamed

### DISCUSSION THREAD

**Discussion by Beefster09** [odin-lang/Odin#2717] [General]
I started using this thinking that this was a pool allocator - i.e. an allocator that operates on equal-size chunks of memory, but as it turns out from what I can see in the source, it seems this is more like a growable arena allocator, but it's not even ideal as a temp allocator because its free_all operation actually frees all blocks instead of just resetting the pool.

I propose that the struct and associated procs should be renamed to Dynamic_Arena or something along those lines.

**Comment by flysand7** [comment]
#4209 Adresses this

**Comment by flysand7** [comment]
The Dynamic Pool is indeed a misnomer. For now the proposed solution is to rename it to dynamic arena, but keep the old aliases for a while.
